### ISSUE TITLE
NVIDIA /etc/mkinitcpio.conf

### ISSUE THREAD

**Issue by ShiGhost** [archlinux/archinstall#1239]
When installing the NVIDIA drivers, you should also add the `nvidia nvidia_modeset nvidia_uvm nvidia_drm` modules to `/etc/mkinitcpio.conf` This is what the Arch Wiki says https://wiki.archlinux.org/title/NVIDIA

**Comment by dylanmtaylor** [comment]
Specifically it states

If you want to ensure it is loaded at the earliest possible occasion, or are noticing startup issues (such as the nvidia kernel module being loaded after the display manager) you can add nvidia, nvidia_modeset, nvidia_uvm and nvidia_drm to the initramfs.

I am not sure that this matters too much, as the driver seems to be functional by the time that the display manager loads up in practice.

**Comment by Torxed** [comment]
It also feels like it's user-specific if you should do this or not?
I don't have it an my nvidia card runs fine. I might be missing some details here tho.

**Comment by dylanmtaylor** [comment]
> It also feels like it's user-specific if you should do this or not? I don't have it an my nvidia card runs fine. I might be missing some details here tho.

On both of my Nvidia devices including a desktop and a laptop I haven't needed to do this sort of configuration.

**Comment by wc7086** [comment]
I don't think it's necessary to add options that have little to no visible performance improvement when the program is working well.

**Comment by Spunkie** [comment]
For people saying they don't experience this issue, this is mostly proprietary nvidia driver + wayland specific. Related issue https://github.com/archlinux/archinstall/issues/1566 

Still, I'm very much opposed to the idea that this isn't something that should be taken care of by archinstall. 

Right now if you run through archinstall and select KDE + proprietary nvidia drivers with anything approaching a modern dGPU. **You end up with a system that by default, crashes on login.** 
___
The user has no indication at all what needs to be done and forces them to comb through the archwiki blindly looking for a solution. On the off chance they figure out they need to load some extra nvidia kernel modules, the user is still being asked to mess with stuff that could easily result in a system that won't even boot. 

Yes the user could _"fix"_ their issue by switching to X11 in sddm, but they have no way of knowing that. And it's besides the point, wayland is here, things like plasma6 and wayland by default are coming. Archinstall can't hide from that 

Ultimately it's a horrible experience that wastes the users time and undermines the point of using archinstall at all.

**Comment by roboboredom** [comment]
I agree with Spunkie. This is a lovely program but there is no harm in doing this, and it can break a few edge case systems if not done.

**Comment by Torxed** [comment]
Apologies for the delay in response btw, been playing a bit of catchup!
